src/feb.c

```diff
@@ -1464,6 +1468,31 @@ int API_FUNC qthread_readFE(aligned_t *restrict       dest,
     return QTHREAD_SUCCESS;
 }                      /*}}} */
 
+/* the way this works is that:
+ * 1 - src's FEB state is ignored
+ * 2 - data is copied from src to destination
+ */
+
+int API_FUNC qthread_readXX(aligned_t *restrict       dest,
+                            const aligned_t *restrict src)
+{                      /*{{{ */
+    qthread_t          *me      = qthread_internal_self();
+
+    assert(qthread_library_initialized);
+
+    if (!me) {
+        return qthread_feb_blocker_func(dest, (void *)src, READXX);
+    }
+    qthread_debug(FEB_CALLS, "dest=%p, src=%p (tid=%i)\n", dest, src, me->thread_id);
+    QTHREAD_FEB_UNIQUERECORD(feb, src, me);
+
+    if (dest && (dest != src)) {
+        *(aligned_t *)dest = *(aligned_t *)src;
+        MACHINE_FENCE;
```

Comment:
Why is there a `MACHINE_FENCE` here? Why not a `MACHINE_FENCE `*before* the copy operation?

I think this gets back to the question of what valid values for readXX are when the value you're reading is (or could be) empty. (see my comment on https://github.com/Qthreads/qthreads/issues/22)